airbyte-integrations/connectors/source-pinterest/source_pinterest/components/auth.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from base64 import standard_b64encode
+from dataclasses import dataclass
+from typing import Any, Mapping
+
+import backoff
+import requests
+from airbyte_cdk.models import FailureType
+from airbyte_cdk.sources.declarative.auth import DeclarativeOauth2Authenticator
+from airbyte_cdk.sources.streams.http.exceptions import DefaultBackoffException
+from airbyte_cdk.utils import AirbyteTracedException
+from airbyte_cdk.utils.airbyte_secrets_utils import add_to_secrets
+
+logger = logging.getLogger("airbyte")
+
+
+@dataclass
+class PinterestOauthAuthenticator(DeclarativeOauth2Authenticator):
```

Comment:
Can you add a comment explaining why we need a custom OAuth authenticator for pinterest instead of the existing low-code auth options

Reply:
Added, thanks